internal/handler/favourite.go

```diff
@@ -0,0 +1,384 @@
+	Favourites []*domain.Favourite `json:"favourites"`
+	Total      int                 `json:"total"`
+	Limit      int                 `json:"limit"`
+	Offset     int                 `json:"offset"`
+}
+
+// TODO: Replace userID extraction from mux.Vars with extraction from request context.
+// This ensures the userID used for the operation is the one from the authenticated token.
+
+// / ListFavourites handles GET /users/{userId}/favourites
+//
+//		@Summary		List user favourites
+//		@Description	Get paginated list of all favourites for a specific user
+//		@Tags			favourites
+//		@Accept			json
+//		@Produce		json
+//	 @Security BearerAuth
+//		@Param			userId	path		string	true	"User ID (UUID)"
+//		@Param			limit	query		int		false	"Number of items per page"	default(20)
+//		@Param			offset	query		int		false	"Number of items to skip"	default(0)
+//		@Param			sortBy	query		string	false	"Sort field"				Enums(created_at, updated_at)
+//		@Param			order	query		string	false	"Sort order"				Enums(asc, desc)
+//		@Success		200		{object}	Response{data=ListFavouritesResponse}
+//		@Failure		400		{object}	InvalidUUIDError
+//		@Failure		404		{object}	NotFoundError
+//		@Failure		500		{object}	InternalServerError
+//		@Router			/users/{userId}/favourites [get]
+func (h *Handler) ListFavourites(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	userID, err := uuid.Parse(vars["userId"])
+	if err != nil {
+		respondError(w, http.StatusBadRequest, err)
+		return
+	}
+
+	// Parse query parameters
+	limit, _ := strconv.Atoi(r.URL.Query().Get("limit"))
+	offset, _ := strconv.Atoi(r.URL.Query().Get("offset"))
+	sortBy := r.URL.Query().Get("sortBy")
+	order := r.URL.Query().Get("order")
+
+	query := domain.NewPageQuery(limit, offset, sortBy, order)
+
+	favourites, total, err := h.service.ListFavourites(r.Context(), userID, query)
+	if err != nil {
+		respondError(w, mapDomainError(err), err)
+		return
+	}
+
+	respondSuccess(w, http.StatusOK, ListFavouritesResponse{
+		Favourites: favourites,
+		Total:      total,
+		Limit:      query.Limit,
+		Offset:     query.Offset,
+	}, "")
+}
+
+// AddFavouriteRequest represents the request to add a favourite
+type AddFavouriteRequest struct {
+	AssetID uuid.UUID `json:"asset_id"`
+}
+
+// AddFavourite handles POST /users/{userId}/favourites
+//
+//		@Summary		Add favourite
+//		@Description	Add an asset to user's favourites
+//		@Tags			favourites
+//		@Accept			json
+//		@Produce		json
+//	 @Security BearerAuth
+//		@Param			userId	path		string				true	"User ID (UUID)"
+//		@Param			request	body		AddFavouriteRequest	true	"Favourite details"
+//		@Success		201		{object}	Response{data=domain.Favourite}
+//		@Failure		400		{object}	BadRequestError
+//		@Failure		404		{object}	NotFoundError
+//		@Failure		409		{object}	ConflictError
+//		@Failure		500		{object}	InternalServerError
+//		@Router			/users/{userId}/favourites [post]
+func (h *Handler) AddFavourite(w http.ResponseWriter, r *http.Request) {
```

Comment:
Could you please explain what will happen if I try to add the same favorite more than once?